supreme/src/commonMain/kotlin/at/asitplus/signum/supreme/Throwables.kt

```diff
@@ -1,6 +1,12 @@
 package at.asitplus.signum.supreme
 
+@RequiresOptIn(message = "Access to potentially hazardous platform-specific internals requires explicit opt-in. Specify @OptIn(HazardousMaterials::class). These accessors are unstable and may change without warning.")
+/** This is an internal property. It is exposed if you know what you are doing. You very likely don't actually need it. */
+annotation class HazardousMaterials
+
 sealed class CryptoException(message: String? = null, cause: Throwable? = null) : Throwable(message, cause)
-class CryptoOperationFailed(message: String) : CryptoException(message)
+open class CryptoOperationFailed(message: String) : CryptoException(message)
+
+open class UnsupportedCryptoException(message: String? = null, cause: Throwable? = null) : CryptoException(message, cause)
 
-class UnsupportedCryptoException(message: String? = null, cause: Throwable? = null) : CryptoException(message, cause)
+class UnlockFailed(message: String? = null, cause: Throwable? = null) : Throwable(message, cause)
```

Comment:
see my other comment. is this an error or auth failed?

Reply:
see updated exception hierarchy